crml/sylo/src/inbox.rs

```diff
@@ -13,57 +13,73 @@
 *     https://centrality.ai/licenses/lgplv3.txt
 */
 
-use frame_support::{decl_module, decl_storage, dispatch::DispatchResult, dispatch::Vec};
-use frame_system::{self, ensure_signed};
-
-pub trait Trait: frame_system::Trait {
-	// add code here
+use frame_support::{decl_error, decl_module, decl_storage, dispatch::DispatchResult, dispatch::Vec, ensure};
+use frame_system::ensure_signed;
+
+const MAX_MESSAGES: usize = 1_000_000;
+const MAX_MESSAGE_LENGTH: usize = 100_000;
+const MAX_DELETE_MESSAGES: usize = 10_000;
+
+type MessageId = u32;
+type Message = Vec<u8>;
+
+pub trait Trait: frame_system::Trait {}
+
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// An inbox cannot store more than MAX_MESSAGES where each message is smaller than MAX_MESSAGE_LENGTH
+		Message,
+		/// Cannot delete more than MAX_DELETE_MESSAGES at a time
+		Delete,
+		/// Cannot assign any more ids to message due to overflow
+		IdOverflow,
```

Comment:
same comments as before - better error names

Reply:
I wanted to keep the errors general but I'll change it to `MessageIdOverflow`